src/Compilers/Core/Portable/Symbols/ISymbolExtensions_IsAccessibleWithin.cs

```diff
@@ -0,0 +1,353 @@
+                case SymbolKind.Alias:
+                    return IsAccessibleWithin(((IAliasSymbol)symbol).Target, within, throughTypeOpt);
+                case SymbolKind.ArrayType:
+                    return IsAccessibleWithin(((IArrayTypeSymbol)symbol).ElementType, within, throughTypeOpt);
+                case SymbolKind.PointerType:
+                    return IsAccessibleWithin(((IPointerTypeSymbol)symbol).PointedAtType, within, throughTypeOpt);
+                case SymbolKind.ErrorType:
+                    return true;
+                case SymbolKind.NamedType:
+                    return IsNamedTypeAccessibleWithin((INamedTypeSymbol)symbol, within, throughTypeOpt);
+                case SymbolKind.TypeParameter:
+                case SymbolKind.Parameter:
+                case SymbolKind.Local:
+                case SymbolKind.Label:
+                case SymbolKind.Namespace:
+                case SymbolKind.DynamicType:
+                case SymbolKind.Assembly:
+                case SymbolKind.NetModule:
+                case SymbolKind.RangeVariable:
+                case SymbolKind.Discard:
+                case SymbolKind.Preprocessing:
+                    // These types of symbols are always accessible (if visible).
+                    return true;
+                case SymbolKind.Method:
+                    var method = (IMethodSymbol)symbol;
+                    if (((IMethodSymbol)symbol).MethodKind == MethodKind.BuiltinOperator)
+                    {
+                        return true;
+                    }
+
+                    goto case SymbolKind.Field;
+
+                case SymbolKind.Property:
+                case SymbolKind.Event:
+                case SymbolKind.Field:
+                    return IsMemberAccessible(symbol.ContainingType, symbol.DeclaredAccessibility, within, symbol.IsStatic ? null : throughTypeOpt);
+
+                default:
+                    throw ExceptionUtilities.UnexpectedValue(symbol.Kind);
+            }
+        }
+
+        private static bool IsNamedTypeAccessibleWithin(INamedTypeSymbol type, ISymbol within, ITypeSymbol throughTypeOpt)
+        {
+            Debug.Assert(within is INamedTypeSymbol || within is IAssemblySymbol);
+            Debug.Assert(type != null);
+
+            if (!type.IsDefinition)
+            {
+                foreach (var typeArg in type.TypeArguments)
+                {
+                    if (!IsAccessibleWithin(typeArg, within, throughTypeOpt))
+                    {
+                        return false;
+                    }
+                }
+            }
+
+            var containingType = type.ContainingType;
+            if (containingType == null)
+            {
+                return IsNonNestedTypeAccessible(type.ContainingAssembly, type.DeclaredAccessibility, within);
+            }
+            else
+            {
+                return IsMemberAccessible(containingType, type.DeclaredAccessibility, within, null);
```

Comment:
nit: consider named arguments for null literals passed as args.  it helps make it clear what that parameter is for.